packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,243 @@
+          padding: mediaQuery.padding.copyWith(bottom: 0),
+          viewPadding: mediaQuery.viewPadding.copyWith(bottom: 0),
+          viewInsets: mediaQuery.viewInsets.copyWith(bottom: 0),
+          systemGestureInsets:
+          mediaQuery.systemGestureInsets.copyWith(bottom: 0),
+          // displayFeatures: mediaQuery.displayFeatures
+          //     .where((e) => e.bounds.overlaps(_pane1Bounds))
+          //     .toList(),
+        ),
+        child: pane1,
+      );
+      _pane2 = MediaQuery(
+        data: mediaQuery.copyWith(
+          // size: _pane2Bounds.size,
+          padding: mediaQuery.padding.copyWith(top: 0),
+          viewPadding: mediaQuery.viewPadding.copyWith(top: 0),
+          viewInsets: mediaQuery.viewInsets.copyWith(top: 0),
+          systemGestureInsets: mediaQuery.systemGestureInsets.copyWith(top: 0),
+          // displayFeatures: mediaQuery.displayFeatures
+          //     .where((e) => e.bounds.overlaps(_pane2Bounds))
+          //     .toList(),
+        ),
+        child: pane2,
+      );
+    } else {
+      _pane1 = MediaQuery(
+        data: mediaQuery.copyWith(
+          // size: _pane1Bounds.size,
+          padding: mediaQuery.padding.copyWith(right: 0),
+          viewPadding: mediaQuery.viewPadding.copyWith(right: 0),
+          viewInsets: mediaQuery.viewInsets.copyWith(right: 0),
+          systemGestureInsets:
+          mediaQuery.systemGestureInsets.copyWith(right: 0),
+          // displayFeatures: mediaQuery.displayFeatures
+          //     .where((e) => e.bounds.overlaps(_pane1Bounds))
+          //     .toList(),
+        ),
+        child: pane1,
+      );
+      _pane2 = MediaQuery(
+        data: mediaQuery.copyWith(
+          // size: _pane2Bounds.size,
+          padding: mediaQuery.padding.copyWith(left: 0),
+          viewPadding: mediaQuery.viewPadding.copyWith(left: 0),
+          viewInsets: mediaQuery.viewInsets.copyWith(left: 0),
+          systemGestureInsets: mediaQuery.systemGestureInsets.copyWith(left: 0),
+          // displayFeatures: mediaQuery.displayFeatures
+          //     .where((e) => e.bounds.overlaps(_pane2Bounds))
+          //     .toList(),
+        ),
+        child: pane2,
+      );
+    }
+
+    return Flex(
+      children: [
+        if (singlePane != TwoPaneSinglePane.pane2)
+          Expanded(
+            flex: (100 * _paneProportion).toInt(),
+            child: _pane1,
+          ),
+        if (singlePane == TwoPaneSinglePane.both)
+          _delimiter,
+        if (singlePane != TwoPaneSinglePane.pane1)
+          Expanded(
+            flex: (100 * (1 - _paneProportion)).toInt(),
+            child: _pane2,
+          ),
+      ],
+      direction: _direction,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      textDirection: _textDirection(context),
+      verticalDirection: verticalDirection,
+    );
+  }
+}
+
+/// Describes which pane to show or if both should be shown.
+enum TwoPaneSinglePane{
```

Comment:
This seems somewhat analogous the to the `TwoPaneView` enum called `TwoPaneViewPriority` - maybe another option for the Flutter name could be `TwoPanePriority`? 
Just a thought (and it's not exactly the same as the other control, because the single/double choice there is a separate property `*ModeConfiguration` which is where you choose whether to show one or both).

Reply:
I see value in having the same naming here so I will change it. Thanks.